modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java

```diff
@@ -54,16 +59,29 @@ protected void configure() {
             waitingFor(
                 new WaitAllStrategy()
                     .withStrategy(waitStrategy)
-                    .withStrategy(Wait.forLogMessage(".*Function worker service started.*", 1))
+                    .withStrategy(createLogWaitingStrategy())
             );
         }
     }
 
+    @Override
+    public PulsarContainer withStartupTimeout(Duration startupTimeout) {
+        this.startupTimeout = startupTimeout;
+        return super.withStartupTimeout(startupTimeout);
+    }
+
     public PulsarContainer withFunctionsWorker() {
         functionsWorkerEnabled = true;
         return this;
     }
 
+    private WaitStrategy createLogWaitingStrategy() {
+        if (startupTimeout != null) {
+            return Wait.forLogMessage(".*Function worker service started.*", 1).withStartupTimeout(startupTimeout);
```

Comment:
We can apply the `startupTimeout` to the `WaitAllStrategy` instead.

Reply:
Done!